src/Runtime/Eval.hs

```diff
@@ -184,15 +185,15 @@ eval (App n es) = do
     Nothing -> do
       f <- lookupName n
       case f of
-        Just (Vf params env' e) -> extScope (zip params (args) ++ env') (evalWithLimit (eval e)) -- ++ env?
-        Just (Pv env' e)        -> extScope (zip [] (args) ++ env') (evalWithLimit (eval e)) -- ++ env?
+        Just (Vf params env' e) -> extScope (extendEvalEnv (params,args) env') (evalWithLimit (eval e))--(Map.union (Map.fromList (zip params (args))) env') (evalWithLimit (eval e)) -- ++ env?
+        Just (Pv env' e)        -> extScope (extendEvalEnv ([],args) env') (evalWithLimit (eval e)) --(Map.union (Map.fromList (zip [] (args))) env') (evalWithLimit (eval e)) -- ++ env?
         Nothing                 -> return $ Err $ "Couldn't find " ++ n ++ " in the environment!"
         _                       -> return $ Err $ n ++ " was not correct when looking it up in the environment!"
 
 -- evaluate a Let expression
 eval (Let n e1 e2) = do
   v <- eval e1
-  extScope (pure (n, v)) (eval e2)
+  extScope (insertEvalEnv (n,v) emptyEvalEnv) (eval e2)
```

Comment:
Here we should use some sort of `newEnv` or `singletonEnv` function that is part of the typeclass.